### DISCUSSION TITLE
Configure agent to generate only certain traces

### DISCUSSION THREAD

**Discussion by ashwin-07** [open-telemetry/opentelemetry-java-instrumentation#7118] [Q&A]
Hello 👋

I exploring java agent for auto instrumentation. The agent generates traces for all events as it should, but I have a use case where I want to exclude most of the API's from tracing.  i.e I want the agent to generate traces and spans only for the functions annotated with `@WithSpan`. I went through the instruction here https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#enable-manual-instrumentation-only. After adding the following properties, no traces get generated even for api's annotated with `@WithSpan `annotation.
```
-Dotel.instrumentation.common.default-enabled=false 
-Dotel.instrumentation.opentelemetry-api.enabled=true 
-Dotel.instrumentation.opentelemetry-annotations.enabled=true
```

Sample Code.
```

@RestController
Class A {

  private final Service1 service;

  @PostMapping
  @WithSpan("save api")
  public ResponseEntity<String> save(@RequestBody String content) {
     var response = service.save(content);
  }

}
```

The save method inside service is also annotated with `@WithSpan`, but no traces gets reported.

Im sure Im missing something here. Can someone advice on what needs to be done to achieve the above scenario?

**Additional Comments**:
I guess manual instrumentation would be suitable here, but I really like the simplicity of setting up the agent, and annotation. And also would not want to be adding much to the codebase. Having an agent makes tracing less code intrusive. So the end goal which I would prefer is to still use agent, but agent should only generate traces and spans for annotated methods. I am also exploring adding custom samplers as extension to agent, but wanted to check if there is any other simple way to achieve this.

**Comment by trask** [comment]
hi @ashwin-07! there was a bug in our docs, which has been fixed now.

in particular,

`-Dotel.instrumentation.opentelemetry-annotations.enabled=true`

should be

`-Dotel.instrumentation.opentelemetry-instrumentation-annotations.enabled=true`
